tracing-subscriber/src/fmt/format.rs

```diff
@@ -280,8 +308,41 @@ where
     }
 }
 
-// === impl FormatFields ===
+#[cfg(feature = "json")]
+impl<N, T> FormatEvent<N> for Format<Json, T>
+where
+    N: for<'writer> FormatFields<'writer>,
+    T: FormatTime,
+{
+    fn format_event(
+        &self,
+        ctx: &span::Context<'_, N>,
+        writer: &mut dyn fmt::Write,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        let mut timestamp = String::new();
+        self.timer.format_time(&mut timestamp)?;
+
+        let mut log_data = json!({
+            "timestamp": timestamp.trim(),
+            "event": event.as_serde(),
+        });
```

Comment:
You might want to replace this with[`SerdeMapVisitor`](https://github.com/tokio-rs/tracing/blob/449550bf0785499265ee7a446347583059135af0/tracing-serde/src/lib.rs#L168-L1710)—which you entail making it `pub`, and using [serde_json's `Serializer` struct](https://docs.serde.rs/serde_json/struct.Serializer.html). Note that `serde_json::Serializer` makes use of `std::io::Write`, while `tracing` uses `std::fmt::Write`. There might be some effort involved in bridging that gap.

Reply:
However, I'm not sure this is _less_ efficient, but it's something I'd like to call out.

Reply:
Reading the rustdocs for both Writes, and it seems if I need to implement a `std::io::Write` to a wrapper that holds a reference to a `std::fmt::Write`, I need to go from `&[u8]` to `&str`. Looks like that needs to go through `String::from_utf8_lossy`, which might allocate in the worst case.

I'll give it a try when I get to the office.

Reply:
Yeah, seems to work nicely :)